build.gradle

```
@@ -76,4 +76,132 @@ project(':configurator') {
+        }
+        
+    }
+}
+
+task unzipJre(description: 'Unzip JREs only when the extract directory is not existing', dependsOn:downloadJre) {
+    doFirst {
+        fileTree(jreDownloadDir).matching {
+            exclude 'jre'
+        }.each { jreFile ->
+                 if(jreFile.name.endsWith('zip')) {
+                     copy {
+                         from zipTree(jreFile)
+                         into jreExtractSubDir('windows')
+                     }
+                 }
+                 else {
+                     copy {
+                         from tarTree(resources.gzip(jreFile))
+                         into jreExtractSubDir(jreFile.name.contains('mac') ? 'mac' : 'linux')
+                     }
+                 }
+        }
+    }
+}
+
+unzipJre.onlyIf {
+    !file(jreExtractDir).exists()
+}
+
+task createPackages(description: 'Create three packages for Linux, Windows and Mac', dependsOn:unzipJre) {
+    //This task also depends on other three tasks which are created by iterating osNames
+}
+
+osNames.each {String sys ->
+    def packageDir = new File("$buildDir/distributions/packages")
+    def (taskType, fileExtension, fileCompression, scriptFile) = setPackageProperties(sys)
+    def taskCLass = Class.forName(taskType)
+    def jarDir = file("$buildDir/libs")
+    def scriptDir = file("$projectDir/launcher-scripts")
+    task "packageFor$sys"(type: taskCLass){
+        mustRunAfter unzipJre
+        archiveBaseName = "admin-console-package-for-$sys"
+        extension = fileExtension
+        if(fileCompression) compression = Compression.GZIP
+        destinationDirectory = packageDir
+        from jarDir
+        from file(jreExtractSubDir(sys))
+        exclude (['*.zip', '*.tar.gz'])
+        from file("${scriptDir.getAbsolutePath()}/$scriptFile")
+
+        doLast {
+            println("Package for $sys is created under ${packageDir.absolutePath}")
+        }
+    }
+    createPackages.dependsOn("packageFor$sys")
+}
+
+def setPackageProperties(String sys) {
+    def scriptFile
+    def taskType
+    def fileExtension
+    def fileCompression
+    def gradleBundlingTaskPackage = { type -> "org.gradle.api.tasks.bundling.$type" }
+    def scriptFileNames = ['Windows-launcher.bat', 'Linux-launcher.sh', 'Mac-launcher.dmg']
+
+    if(sys == osNames[1]){
+        taskType = gradleBundlingTaskPackage('Zip')
+        fileExtension = packageFileExtension[0]
+        fileCompression = false
+        scriptFile = scriptFileNames[0]
+    }
+    else if (sys == osNames[0]){
+        taskType = gradleBundlingTaskPackage('Tar')
+        fileExtension = packageFileExtension[1]
+        fileCompression = true
+        scriptFile = scriptFileNames[1]
+    }
+    else {
+        taskType = gradleBundlingTaskPackage('Tar')
+        fileExtension = packageFileExtension[1]
+        fileCompression = true
+        scriptFile = scriptFileNames[2]
+    }
+    return new Tuple4(taskType,fileExtension,fileCompression,scriptFile)
```

I'd suggest a blank line above this line. And also return is optional, as the value of the last statement of a groovy method is always the return value.

```suggestion

    new Tuple4(taskType,fileExtension,fileCompression,scriptFile)
```